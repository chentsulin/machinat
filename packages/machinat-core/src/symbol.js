// @flow
export const MACHINAT_ELEMENT_TYPE = Symbol('machinat.element');

export const MACHINAT_FRAGMENT_TYPE = Symbol('machinat.element.fragment');

export const MACHINAT_PAUSE_TYPE = Symbol('machinat.element.pause');

export const MACHINAT_PROVIDER_TYPE = Symbol('machinat.element.provider');

export const MACHINAT_THUNK_TYPE = Symbol('machinat.element.thunk');

export const MACHINAT_RAW_TYPE = Symbol('machinat.element.raw');

export const MACHINAT_NATIVE_TYPE = Symbol('machinat.component.native');

export const MACHINAT_SERVICES_CONTAINER = Symbol(
  'machinat.services.container'
);

export const MACHINAT_SERVICES_INTERFACE = Symbol(
  'machinat.services.interface'
);

export const MACHINAT_SERVICES_PROVIDER = Symbol('machinat.services.provider');
