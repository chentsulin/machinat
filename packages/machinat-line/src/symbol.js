export const LINE_NAITVE_TYPE = Symbol.for('machinat.line.native');

export const API_PATH = Symbol.for('machinat.line.api_path');
export const HAS_BODY = Symbol.for('machinat.line.has_body');
export const THREAD_ID = Symbol.for('machinat.line.thread_id');

export const NO_BODY = Symbol.for('machinat.line.act.no_body');
