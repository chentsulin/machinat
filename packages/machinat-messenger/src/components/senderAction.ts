import { unitSegment } from '@machinat/core/renderer';
import type { UnitSegment } from '@machinat/core/renderer/types';
import { annotateMessengerComponent } from '../utils';
import type { MessengerComponent, SenderActionValue } from '../types';

/** @ignore */
const MARK_SEEN_VALUE = { sender_action: 'mark_seen' as const };

/** @ignore */
const TYPING_OFF_VALUE = { sender_action: 'typing_off' as const };

/** @ignore */
const TYPING_ON_VALUE = { sender_action: 'typing_on' as const };

/** @ignore */
const _MarkSeen = function MarkSeen(node, path) {
  return [unitSegment(node, path, MARK_SEEN_VALUE)];
};
/**
 * Display the confirmation icon.
 * @category Component
 * @props {@link LoginButtonProps}
 * @guides Check official [doc](https://developers.facebook.com/docs/messenger-platform/send-messages/sender-actions)
 *   and [reference](https://developers.facebook.com/docs/messenger-platform/reference/send-api/).
 */
export const MarkSeen: MessengerComponent<
  {}, // eslint-disable-line @typescript-eslint/ban-types
  UnitSegment<SenderActionValue>
> = annotateMessengerComponent(_MarkSeen);

/** @ignore */
const _TypingOn = function TypingOn(node, path) {
  return [unitSegment(node, path, TYPING_ON_VALUE)];
};
/**
 * Display the typing bubble.
 * @category Component
 * @props {@link LoginButtonProps}
 * @guides Check official [doc](https://developers.facebook.com/docs/messenger-platform/send-messages/sender-actions)
 *   and [reference](https://developers.facebook.com/docs/messenger-platform/reference/send-api/).
 */
export const TypingOn: MessengerComponent<
  {}, // eslint-disable-line @typescript-eslint/ban-types
  UnitSegment<SenderActionValue>
> = annotateMessengerComponent(_TypingOn);

/** @ignore */
const _TypingOff = function TypingOff(node, path) {
  return [unitSegment(node, path, TYPING_OFF_VALUE)];
};
/**
 * Remove the typing bubble.
 * @category Component
 * @props {@link LoginButtonProps}
 * @guides Check official [doc](https://developers.facebook.com/docs/messenger-platform/send-messages/sender-actions)
 *   and [reference](https://developers.facebook.com/docs/messenger-platform/reference/send-api/).
 */
export const TypingOff: MessengerComponent<
  {}, // eslint-disable-line @typescript-eslint/ban-types
  UnitSegment<SenderActionValue>
> = annotateMessengerComponent(_TypingOff);
