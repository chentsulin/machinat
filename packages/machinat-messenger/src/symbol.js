export const MESSENGER_NAITVE_TYPE = Symbol.for('machinat.messenger.native');
