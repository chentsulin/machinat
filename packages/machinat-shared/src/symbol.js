export const MACHINAT_ELEMENT_TYPE = Symbol.for('machinat.element');

export const MACHINAT_ROOT_ELEMENT = Symbol.for('machinat.element.root');

export const MACHINAT_FRAGMENT_TYPE = Symbol.for('machinat.element.fragment');

export const MACHINAT_ASYNC_TYPE = Symbol.for('machinat.element.async');
