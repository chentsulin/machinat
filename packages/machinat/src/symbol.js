// @flow
export const MACHINAT_ELEMENT_TYPE = Symbol.for('machinat.element');

export const MACHINAT_FRAGMENT_TYPE = Symbol.for('machinat.element.fragment');

export const MACHINAT_PAUSE_TYPE = Symbol.for('machinat.element.pause');

export const MACHINAT_NATIVE_TYPE = Symbol.for('machinat.component.native');

export const MACHINAT_SERVICE_TYPE = Symbol.for('machinat.service');

export const MACHINAT_PROVIDER_TYPE = Symbol.for('machinat.service.provider');

export const MACHINAT_CONSUMER_TYPE = Symbol.for('machinat.service.consumer');

export const SEGMENT_BREAK = Symbol.for('machinat.segment.break');
