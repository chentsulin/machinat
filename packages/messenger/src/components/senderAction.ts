import { makeUnitSegment, UnitSegment } from '@machinat/core/renderer';
import { annotateMessengerComponent } from '../utils';
import type { MessengerComponent, SenderActionValue } from '../types';

const MARK_SEEN_VALUE = { sender_action: 'mark_seen' as const };

const TYPING_OFF_VALUE = { sender_action: 'typing_off' as const };

const TYPING_ON_VALUE = { sender_action: 'typing_on' as const };

/**
 * Display the confirmation icon.
 * @category Component
 * @props `{}`
 * @guides Check official [doc](https://developers.facebook.com/docs/messenger-platform/send-messages/sender-actions)
 *   and [reference](https://developers.facebook.com/docs/messenger-platform/reference/send-api/).
 */
export const MarkSeen: MessengerComponent<
  {},
  UnitSegment<SenderActionValue>
> = annotateMessengerComponent(function MarkSeen(node, path) {
  return [makeUnitSegment(node, path, MARK_SEEN_VALUE)];
});

/**
 * Display the typing bubble.
 * @category Component
 * @props `{}`
 * @guides Check official [doc](https://developers.facebook.com/docs/messenger-platform/send-messages/sender-actions)
 *   and [reference](https://developers.facebook.com/docs/messenger-platform/reference/send-api/).
 */
export const TypingOn: MessengerComponent<
  {},
  UnitSegment<SenderActionValue>
> = annotateMessengerComponent(function TypingOn(node, path) {
  return [makeUnitSegment(node, path, TYPING_ON_VALUE)];
});

/**
 * Remove the typing bubble.
 * @category Component
 * @props `{}`
 * @guides Check official [doc](https://developers.facebook.com/docs/messenger-platform/send-messages/sender-actions)
 *   and [reference](https://developers.facebook.com/docs/messenger-platform/reference/send-api/).
 */
export const TypingOff: MessengerComponent<
  {},
  UnitSegment<SenderActionValue>
> = annotateMessengerComponent(function TypingOff(node, path) {
  return [makeUnitSegment(node, path, TYPING_OFF_VALUE)];
});
