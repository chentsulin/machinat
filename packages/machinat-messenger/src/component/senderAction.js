import { annotate, asNative, asUnit, hasEntry } from 'machinat-utility';

import { MESSENGER_NAITVE_TYPE } from '../symbol';
import { ENTRY_MESSAGES } from '../apiEntry';

const MARK_SEEN_RENDERED = [{ sender_action: 'mark_seen' }];
const TYPING_ON_RENDERED = [{ sender_action: 'typing_on' }];
const TYPING_OFF_RENDERED = [{ sender_action: 'typing_off' }];

export const MarkSeen = () => MARK_SEEN_RENDERED;

annotate(
  asNative(MESSENGER_NAITVE_TYPE),
  hasEntry(ENTRY_MESSAGES),
  asUnit(true)
)(MarkSeen);

export const TypingOn = () => TYPING_ON_RENDERED;

annotate(
  asNative(MESSENGER_NAITVE_TYPE),
  hasEntry(ENTRY_MESSAGES),
  asUnit(true)
)(TypingOn);

export const TypingOff = () => TYPING_OFF_RENDERED;

annotate(
  asNative(MESSENGER_NAITVE_TYPE),
  hasEntry(ENTRY_MESSAGES),
  asUnit(true)
)(TypingOff);
