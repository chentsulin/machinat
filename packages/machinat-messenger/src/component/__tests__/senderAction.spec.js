import Machinat from 'machinat';

import { MESSENGER_NAITVE_TYPE } from '../../symbol';
import { MarkSeen, TypingOn, TypingOff } from '../senderAction';
import renderHelper from './renderHelper';

const render = renderHelper(null);

describe.each([MarkSeen, TypingOn, TypingOff])('%p', Action => {
  it('is valid unit Component', () => {
    expect(typeof Action).toBe('function');
    expect(Action.$$native).toBe(MESSENGER_NAITVE_TYPE);
    expect(Action.$$entry).toBe('me/messages');
    expect(Action.$$unit).toBe(true);
  });

  it('match snapshot', () => {
    expect(render(<Action />)).toMatchSnapshot();
  });
});
