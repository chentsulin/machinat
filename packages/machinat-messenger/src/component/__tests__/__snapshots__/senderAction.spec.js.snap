// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Function MarkSeen] match snapshot 1`] = `
Array [
  Object {
    "sender_action": "mark_seen",
  },
]
`;

exports[`[Function TypingOff] match snapshot 1`] = `
Array [
  Object {
    "sender_action": "typing_off",
  },
]
`;

exports[`[Function TypingOn] match snapshot 1`] = `
Array [
  Object {
    "sender_action": "typing_on",
  },
]
`;
